### PR TITLE
Pointer events tests

### PR DESCRIPTION
**Pull request by luis-valverde-ms**
Tests to verify that pointer events are being raised correctly. This is part of the resolution for #4753 . 

It also includes a minor fix to `PlayModeTestUtilities` so they now also update the left simulated hand when required.

### CODE REVIEW

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+            }
+
+            public State state = new State();
+            private int numFocusing;
+
+            void IMixedRealityFocusChangedHandler.OnBeforeFocusChange(FocusEventData eventData)
+            {
+                Assert.AreNotSame(eventData.OldFocusedObject, eventData.NewFocusedObject);
+                Assert.AreEqual(state.numFocusChanged, state.numBeforeFocusChange);
+
```
Comment (lukastoenneMS) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs]: Can we assert numFocusEnter == numFocusExit here? Or should it be numFocusEnter  == numFocusExit + 1, i.e. already entered?

Similarly i would expect numFocusEnter == numFocusExit + 1 in OnFocusChanged.

Comment (luis-valverde-ms) [reply Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs]: Focus enter/left events are received only for the first pointer entering/last leaving.

Comment (lukastoenneMS) [reply Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs]: I see, never mind then.

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+            // Show right hand far enough from the test collider to not give it focus
+            var rightHand = new Hand(Handedness.Right);
+            Vector3 rightNoFocusPos = new Vector3(0.05f, 0, 0.5f);
+            yield return rightHand.Show(rightNoFocusPos);
+
+            // Make sure no events have been raised
+            var expected = new PointerHandler.State();
+            handler.state.AssertEqual(expected);
+
+            // Show right hand far enough from the test collider to not give it focus
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs]:
```suggestion
            // Show left hand far enough from the test collider to not give it focus
```

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+            {
+                gestureId = newGestureId;
+                return PlayModeTestUtilities.MoveHandFromTo(position, position, 1, gestureId, handedness, simulationService);
+            }
+        }
+
+        // Initializes MRTK, instantiates the test content prefab and adds a pointer handler to the test collider
+        private PointerHandler SetupScene()
+        {
+            TestUtilities.InitializeMixedRealityToolkit(true);
```
Comment (julenka) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs]: Have a look at how lars is setting up and destroying tests in #5033. I think Lars will soon get his changes in, you'll probably need to update setup/teardown to match that pattern.

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.Input;
+
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    // Tests to verify that pointer events are being raised correctly
+    public class PointerEventsTests
+    {
+
+        // Helper script used to keep track of the number of pointer events raised
```
[the remark this answers is not in the document]
Comment (luis-valverde-ms) [reply Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs]: I don't think it is worth using `FocusedObjectEventCatcher<>` here as we're counting events separately and there is not that much more to it.

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs (hunk 3 above)
Comment (luis-valverde-ms) [reply Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs]: We should probably do a batch update of all tests when Lars's changes are in.

**Pull request merged**